docs/bbs/pmo.md

```diff
@@ -0,0 +1,170 @@
+
+| Bit | Count | Description |
+|-----|-------|-------------|
+|  0 | 2 | Texture Coordinate Format |
+|  2 | 3 | Color Format |
+|  5 | 2 | Normal Format (Not used by BBS) |
+|  7 | 2 | Position Format |
+|  9 | 2 | Weight Format |
+| 11 | 2 | Indices Format (Not used by BBS) |
+| 13 | 1 | Unused |
+| 14 | 3 | Number of skinning weights |
+| 17 | 1 | Unused |
+| 18 | 3 | Number of morphing weights (Not used by BBS) |
+| 21 | 2 | Unused |
+| 23 | 1 | Skip Transform Pipline (Not used by BBS) |
+| 24 | 1 | Uniform Diffuse Flag |
+| 25 | 3 | Unknown, possibly unused |
+| 28 | 4 | Primative Type |
+
+Texture Format, Normal Format, Position Format and Weight Format are as follows:
+
+| Value | Meaning |
+|-------|---------|
+| 0x0 | Not Present In Vertex |
+| 0x1 | 8-bit normalized |
+| 0x2 | 16-bit normalized |
+| 0x3 | 32-bit float |
+
+Weight format has only been observed to use 8-bit fixed in actual PMO blobs.
+
+Color Format is as follows:
+
+| Value | Meaning |
+|-------|---------|
+| 0x0 | Not Present In Vertex |
+| 0x4 | 16-bit BGR-5650 |
+| 0x5 | 16-bit ABGR-5551 |
+| 0x6 | 16-bit ABGR-4444 |
+| 0x7 | 32-bit ABGR-8888 |
+
+Only 32-bit ABGR-8888 has been observed in use by actual PMO blobs. If the Uniform Diffuse Flag is set, this value should be 0.
+
+The number of skinning/morphing weights to be read is actually the value in the bit field + 1. If the weight format field is 0, these fields should be ignored.
+
+Index Format has not been seen in actual PMO blobs, but if it is used the expected format is as follows:
+
+| Value | Meaning |
+|-------|---------|
+| 0x0 | Not using indices |
+| 0x1 | 8-bit |
+| 0x2 | 16-bit |
+
+Uniform Diffuse Flag indicates that all vertices in the section should use the same vertex color, which follows the header before the vertex data. See header structure above.
+
+Primative Type is as follows:
+
+| Value | Meaning |
+|-------|---------|
+| 0x0 | Points |
+| 0x1 | Lines |
+| 0x2 | Line Strips |
+| 0x3 | Triangles |
+| 0x4 | Triangle Strips |
+| 0x5 | Triangle Fans |
+| 0x6 | Sprites (Quads) |
+
+Only Triangles and Triangle Strips have been observed in actual PMO blobs.
+
+### Decoding Verts
+
+Once the header has been read, along with any optional fields present, the vertex data immediatly follows.
+Each vertex follows the format described by the vertexFlags field.
+The vertex properties are written in the following order. Note that only position is mandatory.
+
+| Name |
+|------|
+| Joint Weights |
+| Texture Coords |
+| Color |
+| Position |
+
+### Reading 'Normalized' values
+
+Vertex properties which are in the format `8-bit normalized` or `16-bit normalized` need to be converted to floating point values before rendering. Note that position values should read as **signed** values, the others as **unsigned** values. The conversion is as follows:
+
+```c++
```

Comment:
A very small note that will not stop us from merging this: the `c++` is not exactly recognised as the correct one is `cpp`. This is according to Linguist: https://github.com/github/linguist/blob/master/lib/linguist/languages.yml#L600